src/main/java/service/AgendaService.java

```diff
@@ -1,80 +1,126 @@
 package service;
 
+import java.util.Collections;
 import java.util.List;
 
 import model.Course;
 import model.Student;
+import repository.CourseRepository;
+import repository.StudentRepository;
+import repository.TransactionManager;
 
 public class AgendaService {
 
+	private TransactionManager transactionManager;
+	
+	public AgendaService(TransactionManager transactionManager) {
+		this.transactionManager = transactionManager;
+	}
+
 	public List<Student> getAllStudents() {
-		// TODO Auto-generated method stub
-		return null;
+		return transactionManager.studentTransaction(StudentRepository::findAll);
 	}
 
 	public Boolean findStudent(Student student) {
-		// TODO Auto-generated method stub
-		return null;
+		return transactionManager.studentTransaction(studentRepository -> 
+			studentRepository.findById(student.getId()) != null
+		);
 	}
 	
 	public Boolean findCourse(Course course) {
-		// TODO Auto-generated method stub
-		return null;
+		return transactionManager.courseTransaction(courseRepository -> 
+			courseRepository.findById(course.getId()) != null
+		);
 	}
 
 	public void addStudent(Student student) {
-		// TODO Auto-generated method stub
-		
+		transactionManager.studentTransaction(studentRepository -> {
+			if (student != null)
+				studentRepository.save(student);
+			return null;
+		});
 	}
 
 	public void removeStudent(Student student) {
-		// TODO Auto-generated method stub
-		
+		transactionManager.studentTransaction(studentRepository -> {
+			if (student != null)
+				studentRepository.delete(student);
+			return null;
+		});		
 	}
 
 	public void addCourseToStudent(Student student, Course course) {
-		// TODO Auto-generated method stub
-		
+		transactionManager.studentTransaction(studentRepository -> {
+			if (studentRepository.findById(student.getId()) != null)
+				studentRepository.updateStudentCourses(student.getId(), course.getId());
+			return null;
+		});
 	}
 
 	public void removeCourseFromStudent(Student student, Course course) {
-		// TODO Auto-generated method stub
-		
+		transactionManager.studentTransaction(studentRepository -> {
+			if (studentRepository.findById(student.getId()) != null)
+				studentRepository.removeStudentCourse(student.getId(), course.getId());
+			return null;
+		});	
 	}
 
 	public Boolean studentHasCourse(Student student, Course course) {
-		// TODO Auto-generated method stub
-		return null;
+		List<String> studentCourses = transactionManager.studentTransaction(studentRepository -> {
+			if (studentRepository.findById(student.getId()) != null)
+				return studentRepository.findStudentCourses(student.getId());
+			else
+				return Collections.emptyList();
```

Comment:
Calls to "return Collections.emptyList()" inside studentHasCourse() and courseHasStudent() in AgendaService are useless: is this a bug (bad tests) @palai103? They also drop the coverage from Coveralls from 100% to 99.412%.

Reply:
So what do you suggest to replace the return statement with? Because the logic here was to not interact with the `findStudentCourses` repository method at all if not necessary.